aws/resource_aws_sagemaker_domain_test.go

```diff
@@ -0,0 +1,778 @@
+  domain_name = %[1]q
+  auth_mode   = "IAM"
+  vpc_id      = aws_vpc.test.id
+  subnet_ids  = [aws_subnet.test.id]
+
+  default_user_settings {
+    execution_role = aws_iam_role.test.arn
+  }
+}
+`, rName)
+}
+
+func testAccAWSSagemakerDomainConfigSecurityGroup1(rName string) string {
+	return testAccAWSSagemakerDomainConfigBase(rName) + fmt.Sprintf(`
+resource "aws_security_group" "test" {
+  name = "%[1]s"
+}
+
+resource "aws_sagemaker_domain" "test" {
+  domain_name = %[1]q
+  auth_mode   = "IAM"
+  vpc_id      = aws_vpc.test.id
+  subnet_ids  = [aws_subnet.test.id]
+
+  default_user_settings {
+    execution_role  = aws_iam_role.test.arn
+    security_groups = [aws_security_group.test.id]
+  }
+}
+`, rName)
+}
+
+func testAccAWSSagemakerDomainConfigSecurityGroup2(rName string) string {
+	return testAccAWSSagemakerDomainConfigBase(rName) + fmt.Sprintf(`
+resource "aws_security_group" "test" {
+  name = %[1]q
+}
+
+resource "aws_security_group" "test2" {
+  name = "%[1]s-2"
+}
+
+resource "aws_sagemaker_domain" "test" {
+  domain_name = %[1]q
+  auth_mode   = "IAM"
+  vpc_id      = aws_vpc.test.id
+  subnet_ids  = [aws_subnet.test.id]
+
+  default_user_settings {
+    execution_role  = aws_iam_role.test.arn
+    security_groups = [aws_security_group.test.id, aws_security_group.test2.id]
+  }
+}
+`, rName)
+}
+
+func testAccAWSSagemakerDomainConfigTags1(rName, tagKey1, tagValue1 string) string {
+	return testAccAWSSagemakerDomainConfigBase(rName) + fmt.Sprintf(`
+resource "aws_sagemaker_domain" "test" {
+  domain_name = %[1]q
+  auth_mode   = "IAM"
+  vpc_id      = aws_vpc.test.id
+  subnet_ids  = [aws_subnet.test.id]
+
+  default_user_settings {
+    execution_role = aws_iam_role.test.arn
+  }
+
+  tags = {
+    %[2]q = %[3]q
+  }
+}
+`, rName, tagKey1, tagValue1)
+}
+
+func testAccAWSSagemakerDomainConfigTags2(rName, tagKey1, tagValue1, tagKey2, tagValue2 string) string {
+	return testAccAWSSagemakerDomainConfigBase(rName) + fmt.Sprintf(`
+resource "aws_sagemaker_domain" "test" {
+  domain_name = %[1]q
+  auth_mode   = "IAM"
+  vpc_id      = aws_vpc.test.id
+  subnet_ids  = [aws_subnet.test.id]
+
+  default_user_settings {
+    execution_role = aws_iam_role.test.arn
+  }
+
+  tags = {
+    %[2]q = %[3]q
+    %[4]q = %[5]q
+  }
+}
+`, rName, tagKey1, tagValue1, tagKey2, tagValue2)
+}
+
+func testAccAWSSagemakerDomainConfigSharingSettings(rName string) string {
+	return testAccAWSSagemakerDomainConfigBase(rName) + fmt.Sprintf(`
+resource "aws_kms_key" "test" {
+  description             = %[1]q
+  deletion_window_in_days = 7
+}
+
+
+resource "aws_s3_bucket" "test" {
+  bucket        = %[1]q
+  force_destroy = true
+}
+
+resource "aws_sagemaker_domain" "test" {
+  domain_name = %[1]q
+  auth_mode   = "IAM"
+  vpc_id      = aws_vpc.test.id
+  subnet_ids  = [aws_subnet.test.id]
+
+  default_user_settings {
+    execution_role = aws_iam_role.test.arn
+
+    sharing_settings {
+      domain_output_option = "Allowed"
```

Comment:
```
=== CONT  TestAccAWSSagemakerDomain_sharingSettings
    resource_aws_sagemaker_domain_test.go:194: Step 1/2 error: Error running pre-apply refresh: 2021/01/09 15:47:27 [DEBUG] Using modified User-Agent: Terraform/0.12.26 HashiCorp-terraform-exec/0.12.0
        
        Error: Unsupported argument
        
          on config040635969/terraform_plugin_test.tf line 57, in resource "aws_sagemaker_domain" "test":
          57:       domain_output_option = "Allowed"
        
        An argument named "domain_output_option" is not expected here.
        
--- FAIL: TestAccAWSSagemakerDomain_sharingSettings (0.57s)
```

Reply:
Bizzare... tests passed for me i might have messed up some rebase

Reply:
Fixed! Thanks for catching this